### PR TITLE
Problem: the latest SDK is not used

### PR DESCRIPTION
**Pull request by tomtau**
Solution: bumped the Cosmos SDK version, updated the changelog + removed
the broadcast mode hardcoding in CLI (as it can now be configured in app.toml)

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -603,6 +607,8 @@ func GetMaccPerms() map[string][]string {
 }
 
 // initParamsKeeper init params keeper and its subspaces
+// (Amino is still needed for Ledger at the moment)
+// nolint: staticcheck
 func initParamsKeeper(appCodec codec.BinaryMarshaler, legacyAmino *codec.LegacyAmino, key, tkey sdk.StoreKey) paramskeeper.Keeper {
```
[the remark this answers is not in the document]
Comment (tomtau) [reply app/app.go:612]: @sonatype-lift ignore

File: x/chainmain/module.go [hunk 2]
```diff
@@ -115,6 +117,8 @@ func (AppModule) Route() sdk.Route { return sdk.Route{} }
 func (AppModule) QuerierRoute() string { return "" }
 
 // LegacyQuerierHandler returns the capability module's Querier.
+// (Amino is still needed for Ledger at the moment)
+// nolint: staticcheck
 func (am AppModule) LegacyQuerierHandler(*codec.LegacyAmino) sdk.Querier { return nil }
```
[the remark this answers is not in the document]
Comment (tomtau) [reply x/chainmain/module.go:122]: @sonatype-lift ignore

File: x/nft/keeper/querier.go [hunk 3]
```diff
@@ -154,6 +168,8 @@ func queryDenoms(ctx sdk.Context, req abci.RequestQuery, k Keeper, legacyQuerier
 	return bz, nil
 }
 
+// (Amino is still needed for Ledger at the moment)
+// nolint: staticcheck
 func queryNFT(ctx sdk.Context, req abci.RequestQuery, k Keeper, legacyQuerierCdc *codec.LegacyAmino) ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (tomtau) [reply x/nft/keeper/querier.go:173]: @sonatype-lift ignore

File: x/nft/module.go [hunk 4]
```diff
@@ -133,6 +135,8 @@ func (am AppModule) Route() sdk.Route {
 func (AppModule) QuerierRoute() string { return types.RouterKey }
 
 // LegacyQuerierHandler returns the NFT module sdk.Querier.
+// (Amino is still needed for Ledger at the moment)
+// nolint: staticcheck
 func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sdk.Querier {
```
[the remark this answers is not in the document]
Comment (tomtau) [reply x/nft/module.go:140]: @sonatype-lift ignore

File: x/nft/types/codec.go [hunk 5]
```diff
@@ -28,6 +28,8 @@ func init() {
 }
 
 // RegisterLegacyAminoCodec concrete types on codec
+// (Amino is still needed for Ledger at the moment)
+// nolint: staticcheck
 func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
```
[the remark this answers is not in the document]
Comment (tomtau) [reply x/nft/types/codec.go:33]: @sonatype-lift ignore

File: x/supply/module.go [hunk 6]
```diff
@@ -121,6 +123,8 @@ func (AppModule) Route() sdk.Route { return sdk.Route{} }
 func (AppModule) QuerierRoute() string { return types.RouterKey }
 
 // LegacyQuerierHandler returns the capability module's Querier.
+// (Amino is still needed for Ledger at the moment)
+// nolint: staticcheck
 func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sdk.Querier {
```
[the remark this answers is not in the document]
Comment (tomtau) [reply x/supply/module.go:128]: @sonatype-lift ignore

**Review by yihuang — APPROVED** [review]

**Pull request merged**